### ISSUE TITLE
在对一个简单的3分类模型进行预测时，使用load_params和使用load_persistables的预测结果不同

### ISSUE THREAD

**Issue by wang-kangkang** [PaddlePaddle/Paddle#16353]
训练的时候保存模型是这样的
        model_param_path="./model_parameter/params"+str(epoch_id).zfill(3)
        train_param_path="./train_parameter/persistables"+str(epoch_id).zfill(3)
        fluid.io.save_params(executor=exe,dirname=model_param_path,main_program=train_program)
        fluid.io.save_persistables(executor=exe,dirname=train_param_path,main_program=train_program)
测试的时候加载模型时是这样的
    #fluid.io.load_persistables(executor=exe,dirname='../train_parameter/persistables873',main_program=startup_program)
 fluid.io.load_params(executor=exe,dirname='../model_parameter/params873',main_program=startup_program)

**Comment by wang-kangkang** [comment]
上面的现象具体是用load_params方法的3分类准确度0.82，用load_persistables的3分类准确度0.85

而且使用load_params的时候，测试固定的一批图，每次测出来值都不一样。

**Comment by heavengate** [comment]
定位为训练代码可测试代码均使用了optimizer.minimize方法，实际上都跑的是训练

**Comment by wang-kangkang** [comment]
原因找到了，和heavengate说的一样。
错误代码如下：
 optimizer = fluid.optimizer.AdamOptimizer(learning_rate=fluid.layers.exponential_decay(learning_rate=0.001,decay_steps=10000,decay_rate=0.9,staircase=True),regularization=fluid.regularizer.L2DecayRegularizer(regularization_coeff=0.003))
opts = optimizer.minimize(average_loss)
if(phase=='train'):
                return reader,average_loss,img,softmax_out,label
elif(phase=='test'):
                return reader,img,accuracy,softmax_out,label
应该修改成：
if(phase=='train'):
                optimizer = 
 fluid.optimizer.AdamOptimizer(learning_rate=fluid.layers.exponential_decay(learning_rate=0.001,decay_steps=10000,decay_rate=0.9,staircase=True),regularization=fluid.regularizer.L2DecayRegularizer(regularization_coeff=0.003))
                opts = optimizer.minimize(average_loss)
                return reader,average_loss,img,softmax_out,label
elif(phase=='test'):
                return reader,img,accuracy,softmax_out,label
